pkg/kubelet/prober/worker_test.go

```diff
@@ -268,6 +268,48 @@ func TestSuccessThreshold(t *testing.T) {
 	}
 }
 
+func TestStartupProbeSuccessThreshold(t *testing.T) {
+	ctx := context.Background()
+	m := newTestManager()
+	successThreshold := 1
+	failureThreshold := 3
+	w := newTestWorker(m, startup, v1.Probe{SuccessThreshold: int32(successThreshold), FailureThreshold: int32(failureThreshold)})
+	m.statusManager.SetPodStatus(w.pod, getTestNotRunningStatus())
+
+	m.prober.exec = fakeExecProber{probe.Success, nil}
+
+	for i := 0; i < successThreshold+1; i++ {
+		msg := fmt.Sprintf("%d success", successThreshold)
+		expectContinue(t, w, w.doProbe(ctx), msg)
+		expectResult(t, w, results.Success, msg)
+		expectResultRun(t, w, 0, msg)
+	}
```

Comment:
are there any tests we can do down here to prove the prober stopped and doesn't continue? From what I can see, we only test that it continues before the threshold is met, but don't explicitly test it's stopped

Reply:
@haircommander 

Thanks!

In this test case, `successThreshold` is set to 1, so
`i=0`: the `successThreshold` is not met and probe should be executed  
`i=1`(=`successThreshold`): the `successThreshold` is met and probe should not be executed

I've added check whether probe will be executed or not in each case using `w.onHold` with some comments.
If `successThreshold` is exceeded, `w.onHold` turn to true and probe will not be executed anymore.